### PR TITLE
Mkdocs Setup & GitHub Actions powered deploy

### PR DESCRIPTION
**Pull request by paulotruta**
This PR adds a `Mkdocs` project to this repository with a basic configuration ready to be worked on.
It additionally adds a `.gitignore` file, and a GitHub Actions workflow to deploy the resulting site to GitHub Pages.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      contents: write
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.8'
```
Comment (inverse) [inline .github/workflows/deploy.yml]:
```suggestion
          python-version: '3.11'
```

Should we target latest?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      - main
+  pull_request:
+
+jobs:
+  deploy:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
```
Comment (inverse) [inline .github/workflows/deploy.yml:15]: Nice - didnt know about this before!

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  deploy:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
```
Comment (inverse) [inline .github/workflows/deploy.yml:13]: For writing to gh-pages branch right?

**Review by inverse — APPROVED** [review]
Looks like a solid start!

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (paulotruta) [reply .github/workflows/deploy.yml:13]: Exactly ;) I got this example for the action repo: https://github.com/peaceiris/actions-gh-pages

**Pull request merged**